### PR TITLE
fix: simplify managed node example

### PR DESCRIPTION
**Pull request by Zvikan**
### What does this PR do?

- Remove most of managed node group from the example to the node group docs.
- removing external dns from TF plan e2e workflow until domain certificate issue is being addressed.

### Motivation

- Simplifying the example to show one basic and one advanced managed node group.
- Removing hard coded AMI ids from the example

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by kcoleman731 — APPROVED** [review]

**Review by kcoleman731 — APPROVED** [review]

File: docs/node-groups.md [hunk 1]
```diff
@@ -65,6 +65,267 @@ The below example demonstrates advanced configuration options for a managed node
     }
 ```
 
+The below example demonstrates advanced configuration options using Spot/GPU instances/ARM instances/Bottlerocket and custom AMIs managed node groups.
+
+```
```
Comment (bryantbiggs) [inline docs/node-groups.md:70]:
```suggestion
```hcl
```

File: docs/node-groups.md [hunk 2]
```diff
@@ -65,6 +65,267 @@ The below example demonstrates advanced configuration options for a managed node
...
+    #---------------------------------------------------------#
+    spot_m5 = {
+      # 1> Node Group configuration - Part1
+      node_group_name        = "spot-m5"
+      create_launch_template = true              # false will use the default launch template
+      launch_template_os     = "amazonlinux2eks" # amazonlinux2eks  or bottlerocket
+      public_ip              = false             # Use this to enable public IP for EC2 instances; only for public subnets used in launch templates ;
+      pre_userdata           = <<-EOT
+                 yum install -y amazon-ssm-agent
+                 systemctl enable amazon-ssm-agent && systemctl start amazon-ssm-agent"
```
Comment (bryantbiggs) [inline docs/node-groups.md:82]:
```suggestion
                 systemctl enable amazon-ssm-agent && systemctl start amazon-ssm-agent
```

File: docs/node-groups.md [hunk 3]
```diff
@@ -65,6 +65,267 @@ The below example demonstrates advanced configuration options for a managed node
...
+    #---------------------------------------------------------#
+    gpu = {
+      # 1> Node Group configuration - Part1
+      node_group_name        = "gpu-mg5"         # Max 40 characters for node group name
+      create_launch_template = true              # false will use the default launch template
+      launch_template_os     = "amazonlinux2eks" # amazonlinux2eks or bottlerocket
+      public_ip              = false             # Use this to enable public IP for EC2 instances; only for public subnets used in launch templates ;
+      pre_userdata           = <<-EOT
+            yum install -y amazon-ssm-agent
+            systemctl enable amazon-ssm-agent && systemctl start amazon-ssm-agent"
```
Comment (bryantbiggs) [inline docs/node-groups.md:128]:
```suggestion
            systemctl enable amazon-ssm-agent && systemctl start amazon-ssm-agent
```

File: docs/node-groups.md [hunk 4]
```diff
@@ -65,6 +65,267 @@ The below example demonstrates advanced configuration options for a managed node
...
+    #---------------------------------------------------------#
+    arm = {
+      # 1> Node Group configuration - Part1
+      node_group_name        = "arm-mg5"         # Max 40 characters for node group name
+      create_launch_template = true              # false will use the default launch template
+      launch_template_os     = "amazonlinux2eks" # amazonlinux2eks or bottlerocket
+      public_ip              = false             # Use this to enable public IP for EC2 instances; only for public subnets used in launch templates ;
+      pre_userdata           = <<-EOT
+            yum install -y amazon-ssm-agent
+            systemctl enable amazon-ssm-agent && systemctl start amazon-ssm-agent"
```
Comment (bryantbiggs) [inline docs/node-groups.md:171]:
```suggestion
            systemctl enable amazon-ssm-agent && systemctl start amazon-ssm-agent
```

File: docs/node-groups.md [hunk 5]
```diff
@@ -65,6 +65,267 @@ The below example demonstrates advanced configuration options for a managed node
...
+    #---------------------------------------------------------#
+    # Bottlerocket instance type Worker Group
+    #---------------------------------------------------------#
+    bottlerocket_arm = {
+      # 1> Node Group configuration - Part1
+      node_group_name        = "btl-x86"      # Max 40 characters for node group name
+      create_launch_template = true           # false will use the default launch template
+      launch_template_os     = "bottlerocket" # amazonlinux2eks or bottlerocket
+      public_ip              = false          # Use this to enable public IP for EC2 instances; only for public subnets used in launch templates ;
+      pre_userdata           = <<-EOT
```
Comment (bryantbiggs) [inline docs/node-groups.md:255]: same as above - TOML

File: docs/node-groups.md [hunk 6]
```diff
@@ -65,6 +65,267 @@ The below example demonstrates advanced configuration options for a managed node
...
+    #---------------------------------------------------------#
+    # Bottlerocket ARM instance type Worker Group
+    #---------------------------------------------------------#
+    bottlerocket_arm = {
+      # 1> Node Group configuration - Part1
+      node_group_name        = "btl-arm"      # Max 40 characters for node group name
+      create_launch_template = true           # false will use the default launch template
+      launch_template_os     = "bottlerocket" # amazonlinux2eks or bottlerocket
+      public_ip              = false          # Use this to enable public IP for EC2 instances; only for public subnets used in launch templates ;
+      pre_userdata           = <<-EOT
```
Comment (bryantbiggs) [inline docs/node-groups.md:212]: this user data is incorrect for bottlerocket; bottlerocket uses TOML and doesn't have `yum`, `systemctl`, etc.

**Review by bryantbiggs — COMMENTED** [review]
small suggestions - the only thing is the user data for bottlerocket OS which uses TOML and we're showing bash/linux

File: docs/node-groups.md (hunk 6 above)
Comment (vara-bonthu) [reply docs/node-groups.md:212]: Correct. We can remove the `pre_userdata`  for the docs now

**Review by vara-bonthu — COMMENTED** [review]
LGTM! few changes required

File: docs/node-groups.md (hunk 6 above)
Comment (Zvikan) [reply docs/node-groups.md]: Thank you both, this has been addressed, removed pre_userdata from bottlerocket docs

**Comment by Zvikan** [comment]
Thank you @bryantbiggs @vara-bonthu , comments have been addressed.

**Review by bryantbiggs — APPROVED** [review]

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Pull request merged**